crates/base-db/src/input.rs

```diff
@@ -574,23 +686,33 @@ impl CrateGraph {
     pub fn extend(&mut self, mut other: CrateGraph, proc_macros: &mut ProcMacroPaths) {
         let topo = other.crates_in_topological_order();
         let mut id_map: FxHashMap<CrateId, CrateId> = FxHashMap::default();
-
         for topo in topo {
             let crate_data = &mut other.arena[topo];
+
             crate_data.dependencies.iter_mut().for_each(|dep| dep.crate_id = id_map[&dep.crate_id]);
             crate_data.dependencies.sort_by_key(|dep| dep.crate_id);
-
-            let res = self.arena.iter().find_map(
-                |(id, data)| {
-                    if data == crate_data {
-                        Some(id)
-                    } else {
-                        None
+            let res = self.arena.iter().find_map(|(id, data)| {
+                if data.eq_ignoring_origin(crate_data) {
+                    if data.origin.is_lib() && crate_data.origin.is_local() {
```

Comment:
```rs
match (&data.origin, &crate_data.origin) {...}
```
When the origins are the same => simple equivalency check for dependencies
when one of them is a workspace member (is_local) and the other is not (is_lib) => equivalency check with the dev-deps ignored. In this case, if they are considered equivalent, keep the `is_local` one, as the deps may contain dev deps